In [1]:
import numpy as np

In [2]:
data=data=np.genfromtxt("heart.csv", delimiter=",", skip_header=True)

In [3]:
data

array([[63.,  1.,  3., ...,  0.,  1.,  1.],
       [37.,  1.,  2., ...,  0.,  2.,  1.],
       [41.,  0.,  1., ...,  0.,  2.,  1.],
       ...,
       [68.,  1.,  0., ...,  2.,  3.,  0.],
       [57.,  1.,  0., ...,  1.,  3.,  0.],
       [57.,  0.,  1., ...,  1.,  2.,  0.]])

# Question: What is the relationship between age, sex, trestbps and cholesterol? 

In [4]:
age=data[:,0]

In [5]:
sex=data[:,1]

In [6]:
chol=data[:,4]

In [7]:
trest=data[:,3]

In [25]:
cp=data[:,2]

# Descriptive Statistics

In [8]:
# age
print("---Basic Descriptive statistics on Age---")
print(f"Mean: {round(np.mean(age))}")
print(f"Median: {round(np.median(age))}")
print(f"Standard Deviation: {round(np.std(age))}")
print(f"Minimum: {np.min(age)}")
print(f"Maximum: {np.max(age)}")

---Basic Descriptive statistics on Age---
Mean: 54
Median: 55
Standard Deviation: 9
Minimum: 29.0
Maximum: 77.0


In [9]:
# trest
print("---Basic Descriptive statistics on Trestbps---")
print(f"Mean: {round(np.mean(trest))}")
print(f"Median: {round(np.median(trest))}")
print(f"Standard Deviation: {round(np.std(trest))}")
print(f"Minimum: {np.min(trest)}")
print(f"Maximum: {np.max(trest)}")

---Basic Descriptive statistics on Trestbps---
Mean: 132
Median: 130
Standard Deviation: 18
Minimum: 94.0
Maximum: 200.0


In [10]:
# chol
print("---Basic Descriptive statistics on Cholestrol---")
print(f"Mean: {round(np.mean(chol))}")
print(f"Median: {round(np.median(chol))}")
print(f"Standard Deviation: {round(np.std(chol))}")
print(f"Minimum: {np.min(chol)}")
print(f"Maximum: {np.max(chol)}")

---Basic Descriptive statistics on Cholestrol---
Mean: 246
Median: 240
Standard Deviation: 52
Minimum: 126.0
Maximum: 564.0


Data Filtering

In [11]:
print(f"Number of rows before filtering: {len(data)}")

Number of rows before filtering: 303


In [32]:
male_data=data[(sex==1)]
print(f"Number of rows after filtering for males: {len(male_data)}")

Number of rows after filtering for males: 207


In [33]:
female_data=data[(sex==0)]
print(f"Number of rows after filtering for females: {len(female_data)}")

Number of rows after filtering for females: 96


In [15]:
# High Chol is >=200
high_chol=data[(chol>=200)]
print(f"The number of people who have high cholestrol: {len(high_chol)}")

The number of people who have high cholestrol: 253


In [34]:
high_chol_man=data[(chol>=200) & (sex==1)]
print(f"The number of males with high cholestrol: {len(high_chol_man)}")

The number of males with high cholestrol: 171


In [35]:
high_chol_female=data[(chol>=200) & (sex==0)]
print(f"The number of females with high cholestrol: {len(high_chol_female)}")

The number of females with high cholestrol: 82


In [19]:
# high_trestbps >=140
high_trest=data[(trest>=140)]
print(f"The number of people who have high cholestrol: {len(high_trest)}")

The number of people who have high cholestrol: 97


In [36]:
high_trest_male=data[(trest>=140) & (sex==1)]
print(f"The number of males who have high cholestrol: {len(high_trest_male)}")

The number of males who have high cholestrol: 65


In [37]:
high_trest_female=data[(trest>=140) & (sex==0)]
print(f"The number of females who have high cholestrol: {len(high_trest_female)}")

The number of females who have high cholestrol: 32


In [40]:
#Chest Pain: 
# 3 = typical angina (angina constant chest pain/discomfort that is a symptom of cornaey cardiac disease)

high_cp=data[(cp==3)]
print(f"The number of people who have angina: {len(high_cp)}")
low_cp=data[(cp<3)]
print(f"The number of people who do not have angina: {len(low_cp)}")
male_cp=data[(cp==3) & (sex==1)]
print(f"The number of males who have angina: {len(male_cp)}")
female_cp=data[(cp==3) & (sex==0)]
print(f"The number of females who have angina: {len(female_cp)}")

The number of people who have angina: 23
The number of people who do not have angina: 280
The number of males who have angina: 19
The number of females who have angina: 4


In [50]:
# Data Filtering for Age
old_age=data[(age>=50)]
print(f"The number of people over 50 are: {len(old_age)}")
young_age=data[(age<50)]
print(f"The number of people younger than 50 are: {len(young_age)}")
old_age_chol=data[(age>=50 ) & (chol>=200)]
print(f"The number of people over 50 and have high chol are are: {len(old_age_chol)}")

The number of people over 50 are: 215
The number of people younger than 50 are: 88
The number of people over 50 and have high chol are are: 187


# Correlation

In [43]:
# Age & Cholestrol

corr_age_chol=np.corrcoef(age,chol,rowvar=False)
print(f"Corr age and chol is: {corr_age_chol}")

Corr age and chol is: [[1.         0.21367796]
 [0.21367796 1.        ]]


In [44]:
# Age & Trestbps

corr_age_trest=np.corrcoef(age,trest,rowvar=False)
print(f"Corr age and trest is: {corr_age_trest}")

Corr age and trest is: [[1.         0.27935091]
 [0.27935091 1.        ]]


In [45]:
# Age & Chest Pain

corr_age_cp=np.corrcoef(age,cp,rowvar=False)
print(f"Corr age and chest pain is: {corr_age_cp}")

Corr age and chest pain is: [[ 1.         -0.06865302]
 [-0.06865302  1.        ]]


In [47]:
# Cholestrol & Trestbps

corr_chol_trest=np.corrcoef(chol,trest,rowvar=False)
print(f"Corr cholestrol and trestbps is: {corr_chol_trest}")

Corr cholestrol and trestbps is: [[1.         0.12317421]
 [0.12317421 1.        ]]


In [48]:
# Cholestrol & Chest pain

corr_chol_cp=np.corrcoef(chol,cp,rowvar=False)
print(f"Corr cholestrol and chest pain is: {corr_chol_cp}")

Corr cholestrol and chest pain is: [[ 1.         -0.07690439]
 [-0.07690439  1.        ]]


In [49]:
# Trest BPS & Chest pain

corr_trest_cp=np.corrcoef(cp,trest,rowvar=False)
print(f"Corr cholestrol and chest pain is: {corr_trest_cp}")

Corr cholestrol and chest pain is: [[1.         0.04760776]
 [0.04760776 1.        ]]


# Summary

### We used age, sex, cholestrol, trestbps, and chestpain as our vaules. We used descriptive statistics, filtered data, and correlation. For this data set, it was more useful to use data filtering rather than correlation. We found little to no correlation between any of our five variables. However, we did discover, via filtering that more men had high cholesterol, trest, and chest pain then women. Howver, there were many more men in this study. So, we decided to not just look at the raw number, but the percentage by gender of high cholesterol. 82% of men in the study had high cholestrol. Whereas, 85% of women had high cholestrol.